testng-core/src/test/kotlin/org/testng/internal/ClassHelperTest.kt

```diff
@@ -0,0 +1,143 @@
+import org.testng.internal.samples.classhelper.misamples.BatmanSample
+import org.testng.internal.samples.classhelper.misamples.JohnTravoltaMovesSample
+import org.testng.internal.samples.classhelper.misamples.MickJaggerSample
+import org.testng.xml.XmlClass
+import org.testng.xml.XmlSuite
+import org.testng.xml.XmlTest
+
+class ClassHelperTest {
+
+    @Test
+    fun testGetAvailableMethods() {
+        runTest(getExpected(), LittlePandaSample::class.java)
+    }
+
+    @Test
+    fun testGetAvailableMethodsWhenOverridingIsInvolved() {
+        val expected = getExpected("equals", "hashCode", "toString")
+        runTest(expected, BabyPandaSample::class.java)
+    }
+
+    @Test
+    fun testFindClassInSameTest() {
+        runTest(TestClassSample::class.java, 1, TestClassSample::class.java)
+    }
+
+    @Test
+    fun testFindClassesInSameTest() {
+        runTest(
+            TestClassSample::class.java, 2, TestClassSample::class.java,
+            BabyPandaSample::class.java
+        )
+    }
+
+    @Test
+    fun testNoClassDefFoundError() {
+        val urlClassLoader = object : URLClassLoader(emptyArray<URL>()) {
+            override fun loadClass(className: String?): Class<*> {
+                throw NoClassDefFoundError()
+            }
+        }
+        ClassHelper.addClassLoader(urlClassLoader)
+        val fakeClassName = UUID.randomUUID().toString()
+        assertThat(ClassHelper.forName(fakeClassName))
+            .withFailMessage("The result should be null; no exception should be thrown.")
+            .isNull()
+    }
+
+    @Test(dataProvider = "data")
+    fun testWithDefaultMethodsBeingOverridden(
+        cls: Class<*>, expectedCount: Int, vararg expected: String
+    ) {
+        val methods = ClassHelper.getAvailableMethodsExcludingDefaults(cls)
+        assertThat(methods).hasSize(expectedCount)
+        methods.forEach {
+            val actual = it.declaringClass.name + "." + it.name
+            val isPresent = listOf(*expected).contains(actual)
+            assertThat(isPresent).isTrue()
+        }
+    }
+
+    @DataProvider(name = "data")
+    fun getTestData(): Array<Array<Any>> {
+        return arrayOf(
+            arrayOf(MickJaggerSample::class.java, 1, MickJaggerSample::class.java.name + ".dance"),
+            arrayOf(
+                JohnTravoltaMovesSample::class.java,
+                2,
+                JohnTravoltaMovesSample::class.java.name + ".walk",
+                AbstractMovesSample::class.java.name + ".dance"
+            ),
+            arrayOf(
+                BatmanSample::class.java,
+                3,
+                BatmanSample::class.java.name + ".fly",
+                BatmanSample::class.java.name + ".liftWeights",
+                BatmanSample::class.java.name + ".yellSlogan"
+            )
+        )
+    }
+
+    fun runTest(classToBeFound: Class<*>, expectedCount: Int, vararg classes: Class<*>) {
```

Comment:
Use a data provider instead of individual tests.

Reply:
Fixed